compiler/sem.nim

```diff
@@ -158,6 +158,8 @@ proc commonType*(x, y: PType): PType =
       b = b.lastSon.skipTypes({tyGenericInst})
     if a.kind == tyObject and b.kind == tyObject:
       result = commonSuperclass(a, b)
+      # bug #7818, put this into commonSuperclass?
+      if k in {tyRef, tyPtr}: result = skipTypes(result, skipPtrs)
```

Comment:
This is still not correct. ``skipTypes`` skips every pointer indirection but the subtype relation only holds for a single indirection.